src/NodaTime/LocalDate.cs

```diff
@@ -506,10 +508,16 @@ public static LocalDate FromYearMonthWeekAndDay(int year, int month, int occurre
         /// later than <paramref name="other"/>.</returns>
         public int CompareTo(LocalDate other)
         {
-            Preconditions.CheckArgument(Calendar.Equals(other.Calendar), nameof(other), "Only values with the same calendar system can be compared");
-            return Calendar.Compare(YearMonthDay, other.YearMonthDay);
+            Preconditions.CheckArgument(CalendarOrdinal == other.CalendarOrdinal, nameof(other), "Only values with the same calendar system can be compared");
+            return TrustedCompareTo(other);
         }
 
+        /// <summary>
+        /// Performs a comparison with another date, trusting that the calendar of the other date is already correct.
+        /// This avoids duplicate calendar checks.
+        /// </summary>
+        private int TrustedCompareTo([Trusted] LocalDate other) => Calendar.Compare(YearMonthDay, other.YearMonthDay);
```

Comment:
(Why is `Compare()` an instance method on `CalendarSystem`? Shouldn't it be static?)

Reply:
Nope, it depends on the calendar system. The Hebrew scriptural calendar system prevents the straightforward comparison you might expect.

Reply:
Ah, sorry, too much Java lately: I misread `return YearMonthDayCalculator.Compare()` in the implementation as a static method call on `YearMonthDayCalculator`, not a call through the public member property (that's also called `YearMonthDayCalculator`).